addons/website/models/website.py

```diff
@@ -477,23 +477,31 @@ def set_images(images):
 
         website = self.get_current_website()
         theme_name = kwargs['theme_name']
-        theme = self.env['ir.module.module'].search([('name', '=', theme_name)])
+        theme = request.env['ir.module.module'].search([('name', '=', theme_name)])
         url = theme.button_choose_theme()
+        # button_choose_theme redirects the user with
+        # with_loading and enable_editor but if the
+        # user is on a translated page (e.g /en)
+        # the editor never starts (as there's no edit button)
+        # thus never removes the loading animation
+        # so we redirect the user to the default 
+        # language so the editor can initialize properly
+        url['url'] = '/website/lang/default?r=%s' % urls.url_quote_plus(url['url'])
```

Comment:
:+1: 

Should use `url_encode` though, that's what we normally use unless there are specific reasons not to.